Automattic/wp-calypso
# Improve locale state in calypso

This PR seeks to improve how locale is set or updated by moving the value from the context to the redux state.

#### Testing instructions
  
1. Run `git checkout update/lang` and start your server, or open a [live branch](https://calypso.live/?branch=update/lang)
2. Open the [`Login` page](http://calypso.localhost:3000/log-in/fr)
3. Assert that you see the correct language & no CSS is loaded twice
4. Assert that correct (RTL/LTR) CSS tags are received from the server ( view-source )

| User logged in | User language | URL Language Param | Result language |
|----------------|---------------|--------------------|-----------------|
| ✓              | HE            | FR                 | FR              |
| ✓              | HE            | HE                 | HE              |
| ✓              | FR            | FR                 | FR              |
| ✓              | HE            | N/A                | HE              |
| ✓              | FR            | N/A                | FR              |
| ✗              | N/A           | HE                 | HE              |
| ✗              | N/A           | FR                 | FR              |
| ✗              | N/A           | N/A                | EN              |
 
5. Assert that the Support User feature still works as intended and that you are switched to the user's locale.

* Note: Each test should be performed with and without `"wpcom-user-bootstrap": true`. See #18264 
#### Additional notes

* Commit [d54d453](https://github.com/Automattic/wp-calypso/pull/18297/commits/d54d4533df989969df678132195dbae6c1e5e8fd) is only for testing and should be removed before merging.
* SSR is only used for the default locale ( EN ).

#### Reviews
